_rules/target-size-enhanced-gi8qkf.md

```diff
@@ -0,0 +1,819 @@
+---
+id: gi8qkf
```

Comment:
For first pass of review, please focus on global structure to se whether this is a good structure for the rule. We'll discuss details of the definitions later.